src/pages/StocktakeEditPage.js

```diff
@@ -1,564 +1,224 @@
-import { GenericChoiceList, PageButton, PageInfo, TextEditor } from '../widgets';
-import { PageContentModal, ConfirmModal, StocktakeBatchModal } from '../widgets/modals';
+import { MODAL_KEYS } from '../utilities';
+import { buttonStrings } from '../localization';
+import { DataTablePageModal } from '../widgets/modals';
+import { PageButton, PageInfo } from '../widgets';
+import { DataTable, DataTableHeaderRow, DataTableRow } from '../widgets/DataTable';
+import DataTablePageView from './containers/DataTablePageView';
 
-import { parsePositiveInteger, truncateString, sortDataBy } from '../utilities';
-import { SUSSOL_ORANGE } from '../globalStyles';
 import {
-  buttonStrings,
-  modalStrings,
-  navStrings,
-  tableStrings,
-  pageInfoStrings,
-  programStrings,
-} from '../localization';
-
-const DATA_TYPES_SYNCHRONISED = ['StocktakeItem', 'StocktakeBatch', 'ItemBatch', 'Item'];
-
-const MODAL_KEYS = {
-  COMMENT_EDIT: 'commentEdit',
-  NAME_EDIT: 'nameEdit',
-  REASON_EDIT: 'reasonEdit',
-};
-
-// Below functions are redundant in stocktake now, as stock movement check when opening the
-// stocktake should ensure snapshot is never an old value, making it not possible to add
-// negative adjustments.
-//
-// This is left here for now as a fail safe.
-const formatErrorItemNames = items => {
-  const MAX_ITEMS_IN_ERROR_MESSAGE = 4; // Number of items to display in finalise error modal.
-  const MAX_ITEM_STRING_LENGTH = 40; // Length of string representing item in error modal.
-  let itemsString = '';
-  items.forEach((item, index) => {
-    if (!item) return; // Rerendering sometimes causes a crash here.
-    if (index >= MAX_ITEMS_IN_ERROR_MESSAGE) return;
-    itemsString += truncateString(`\n${item.itemCode} - ${item.itemName}`, MAX_ITEM_STRING_LENGTH);
-  });
-  if (items.length > MAX_ITEMS_IN_ERROR_MESSAGE) {
-    itemsString +=
-      `\n${modalStrings.and} ` +
-      `${items.length - MAX_ITEMS_IN_ERROR_MESSAGE} ` +
-      `${modalStrings.more}.`;
-  }
-  return itemsString;
-};
+  filterData,
+  selectRow,
+  deselectRow,
+  closeBasicModal,
+  editComment,
+  focusNext,
+  focusCell,
+  sortData,
+  editCountedTotalQuantity,
+  closeStocktakeBatchModal,
+  openModal,
+  resetStocktake,
+} from './dataTableUtilities/actions';
+import { recordKeyExtractor, getItemLayout } from './dataTableUtilities/utilities';
+import { SUSSOL_ORANGE, newDataTableStyles, newPageStyles } from '../globalStyles';
+import usePageReducer from '../hooks/usePageReducer';
+import { gotoStocktakeManagePage } from '../navigation/actions';
 
 /**
- * Check whether a given stocktake is safe to be finalised.
- * If stocktake is safe to finalise, return null, else return an appropriate error
- * message.
+ * Renders a mSupply page with a stocktake loaded for editing
  *
- * @param  {object}  stocktake  The stocktake to check.
- * @return {string}             Null if safe to be finalised, else an error message.
- */
-export const checkForFinaliseError = stocktake => {
```

Comment:
This didn't get shifted to `utilities/finalisation` like the other finalise methods